src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -246,6 +252,16 @@ public void AddHooks(IEnumerable<Hook> hooks)
             {
                 var contextFromHooks = await this.TriggerBeforeHooksAsync(allHooks, hookContext, options, cancellationToken).ConfigureAwait(false);
 
+                // short circuit evaluation entirely if provider is in a bad state

```

Comment:
This is probably another comment outside the scope of this PR, but similar to the provider events from init this creates some potentially interesting problems. 

An SDK can fail to initialize, but be capable of handling evaluations. (The LD SDK, for instance, will always evaluate the flag and does other work associated with that evaluation even if it returns a default value.)

When the provider is bypassed like this it means that the analytic events associated with those evaluations are also bypassed. (Because a failure to be able to retrieve configuration is not 100% correlated with not being able to make other connections).

Reply:
I agree fully, this is one of the reasons we don't bypass the `ERROR` state though. We only bypass evaluations if the provider has not been initialized (`NOT_READY`) or if it's in `FATAL` state.

In both these situations, I don't think analytics events could even be reliably transmitted anyway, since we likely haven't connected or there's a credential error, or something the like.

Reply:
This would also apply to basic non-blocking cases though, correct? Set the provider without waiting and start evaluating flags? Typically I would expect a vendor to cache events and then send them once the provider is ready.

Reply:
Regardless though. It is fine for this PR.